File: C++ Programming/Competitive Programming/Coding Challenges 7/program2.cpp
//Q2.Write a program which accept number from user and print even factors of that number.
//Input : 24
//Output: 1 2 4 6 8 12

#include <iostream>
using namespace std;

void DisplayFactor(int iNo)
{
    int i = 0;

    if (iNo <= 0)
    {
        iNo = -iNo;
    }

    for (i = 1; i <= iNo / 2; i++)
    {
        if ((iNo % i == 0) && ((i % 2 == 0) || (i == 1)))
        {
            cout << i << "\t";
        }
    }
}

int main()
{
    int iValue = 0;

    cout << "Enter number: ";
    cin >> iValue;

    DisplayFactor(iValue);

    return 0;
}

File: Java Programming/Competitive Programming/Coding Challenges 15/program2.java
/*
2.Write a program which accept number from user and return the count of odd
digits.

Input : 2395
Output : 3

Input : 1018
Output : 2

Input : -1018
Output : 2

Input : 8462
Output : 0
*/

import java.util.Scanner;

class program2
{
    static int CountOdd(int iNo)
    {
        int iCount = 0;

        String str = Integer.toString(iNo);

        for (int i = 0; i < str.length(); i++)
        {
            char ch = str.charAt(i);

            if (ch >= '0' && ch <= '9')
            {
                int iDigit = ch - '0';

                if (iDigit % 2 != 0)
                {
                    iCount++;
                }
            }
        }

        return iCount;
    }

    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;
        int iRet = 0;

        System.out.print("Enter number : ");
        iValue = sobj.nextInt();

        iRet = CountOdd(iValue);

        System.out.print("Count of odd digits is : " + iRet);

        sobj.close();
    }
}